File: DAILY_CHALLENGE/MARCH_2024/24_03_2024.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        n,max_idx,curr_max=len(nums),0,0

        for i in range(0,n):
            id=nums[i]%n
            nums[id]+=n

        for i in range(0,n):

            if(nums[i]>curr_max):
                curr_max=nums[i]
                max_idx=i
            
            nums[i]%=n
        
        return max_idx

File: DAILY_CHALLENGE/MARCH_2024/24_03_2024.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n=nums.size(),max_idx=0,curr_max=0;
        for(int i=0;i<n;i++)
        {
            int id=nums[i]%n;
            nums[id]+=n;
        }
        for(int i=0;i<n;i++)
        {
            if(nums[i]>curr_max)
            {
                curr_max=nums[i];
                max_idx=i;

            }
            nums[i]%=n;
        }
        return max_idx;
    }
};
